### PR TITLE
Change to use eks.nodeGroupOptions in the builder

### PR DESCRIPTION
**Pull request by ROunofF**
*Issue #, if available:*

*Description of changes:* Change to use eks.nodeGroupOptions so we can have all options override in case it's needed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/builders/windows-builder.ts [hunk 1]
```diff
@@ -191,47 +186,53 @@ class UsageTrackingAddOn extends NestedStack {
...
-    const result : clusterproviders.ManagedNodeGroup = {
-        id: "mng-windows",
-        amiType: options.windowsAmiType,
-        instanceTypes: [new ec2.InstanceType(`${options.instanceClass}.${options.instanceSize}`)],
-        desiredSize: options.desiredNodeSize, 
-        minSize: options.minNodeSize, 
-        maxSize: options.maxNodeSize,
-        nodeRole: options.nodeRole,
-        nodeGroupSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS },
-        diskSize: options.blockDeviceSize,
```
Comment (elamaran11) [inline lib/builders/windows-builder.ts:232]: Why is this property removed?

File: lib/builders/windows-builder.ts [hunk 2]
```diff
@@ -191,47 +186,53 @@ class UsageTrackingAddOn extends NestedStack {
...
  * @param: options: WindowsOptions
  * @returns: blueprints.ManagedNodeGroup
  */
-function addGenericNodeGroup(options: WindowsOptions): clusterproviders.ManagedNodeGroup {
+function buildGenericNodeGroup(options: WindowsOptions, overrideOptions?: eks.NodegroupOptions): clusterproviders.ManagedNodeGroup {
+
+    let currentOptions = options.genericNodeGroupOptions;
+    if ( overrideOptions ) { currentOptions = merge(options.genericNodeGroupOptions, overrideOptions) }
 
     return {
```
Comment (elamaran11) [inline lib/builders/windows-builder.ts:215]: Please fix the GH action issue

**Review by elamaran11 — COMMENTED** [review]
@ROunofF Minor feedback. Overall looks good.

File: lib/builders/windows-builder.ts [hunk 3]
```diff
@@ -191,47 +186,53 @@ class UsageTrackingAddOn extends NestedStack {
...
+/**
+ * This function adds a generic node group to the cluster.
  * @param: options: WindowsOptions
  * @returns: blueprints.ManagedNodeGroup
  */
-function addGenericNodeGroup(options: WindowsOptions): clusterproviders.ManagedNodeGroup {
+function buildGenericNodeGroup(options: WindowsOptions, overrideOptions?: eks.NodegroupOptions): clusterproviders.ManagedNodeGroup {
+
+    let currentOptions = options.genericNodeGroupOptions;
+    if ( overrideOptions ) { currentOptions = merge(options.genericNodeGroupOptions, overrideOptions) }
```
Comment (elamaran11) [inline lib/builders/windows-builder.ts:213]: Missing semicolon

**Review by elamaran11 — COMMENTED** [review]
One more comment.

**Review by shapirov103 — COMMENTED** [review]
@ROunofF please address the minor feedback (inline) I will test and merge.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**